File: practice/data_structures/arrays/dynamic_array/dynamic_array.py
#!/usr/bin/env python

n, q = map(int, raw_input().strip().split())

seq = [[] for _ in range(n)]
last = 0
for _ in xrange(q):
    a, x, y = map(int, raw_input().strip().split())
    pos = (x ^ last) % n
    if a == 1:
        seq[pos].append(y)
    else:
        last = seq[pos][y % len(seq[pos])]
        print last

File: practice/data_structures/dynamic_array/dynamic_array.cpp
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int main()
{
    int n, q;
    int a, x, y;
    int last = 0;
    
    cin >> n >> q;
    
    vector<vector<int>> seq(n, vector<int>());
    while (q--)
    {
        cin >> a >> x >> y;
        
        int pos = (x ^ last) % n;
        if (a == 1)
        {
            seq[pos].push_back(y);
        }
        else
        {
            last = seq[pos][y % seq[pos].size()];
            cout << last << endl;
        }
    }
    return 0;
}

